rust/crd/src/lib.rs

```diff
@@ -326,18 +326,18 @@ impl SparkApplication {
                     .with_config_map(log_config_map)
                     .build(),
             );
-
-            result.push(
-                VolumeBuilder::new(VOLUME_MOUNT_NAME_LOG)
-                    .with_empty_dir(
-                        None::<String>,
-                        Some(product_logging::framework::calculate_log_volume_size_limit(
-                            &[MAX_SPARK_LOG_FILES_SIZE, MAX_INIT_LOG_FILES_SIZE],
-                        )),
-                    )
-                    .build(),
-            );
         }
+        // This volume is also used by the containerdebug process so it must always be there.
+        result.push(
+            VolumeBuilder::new(VOLUME_MOUNT_NAME_LOG)
+                .with_empty_dir(
+                    None::<String>,
+                    Some(product_logging::framework::calculate_log_volume_size_limit(
+                        &[MAX_SPARK_LOG_FILES_SIZE, MAX_INIT_LOG_FILES_SIZE],
+                    )),
+                )
+                .build(),
+        );
```

Comment:
I see that this was added unconditionally to the volume mounts for both variants, but I don't see a removed conditional mount anywhere.

Reply:
Yep, it's also mounted in `add_common_volume_mounts` _if_ `logging_enabled`.

Reply:
I guess this also raises questions about what it means to disable logging here... Would it make more sense to disable containerdebug in those cases, or to disable the vector output at least? Hmm...

Reply:
It get's slightly weirder

Reply:
Huh.. I'm starting to think someone more involved with this operator should have a hand in this review.. >.<

Reply:
> I guess this also raises questions about what it means to disable logging here

This gave me a serious scare.

Looking at other operators (for zookeeper and hdfs) I see that the `log` volume and `/stackable/log` mount are always there regardless of the "state of logging".
As I understand, "logging enabled" just means that _the vector agent is running_. @siegfriedweber am I correct?

> Yep, it's also mounted in add_common_volume_mounts if logging_enabled.

This my fault and stems from the fact that the `submit` role has no `logging` property but the `driver` and `executor` do. This was intentional but probably the wrong decision.

Reply:
> As I understand, "logging enabled" just means that _the vector agent is running_. @siegfriedweber am I correct?

In #355, logging to files was removed for the Spark submit job. So, "logging enabled" in this context means, that it is enabled always for the driver and executer (hence the static "true" parameter) and that it is always disabled for the submit job (the static "false" parameter).

Reply:
Well there's not much of a point in enabling file output if there's no vector there to pick it up.

Reply:
The idea was, that customers can add a sidecar container themselves to pick up the logs with their desired technology.

This is actually done by one customer who uses a custom sidecar container to read the OPA decision logs and forwards them directly via syslog to their SIEM solution.

Reply:
> In #355, logging to files was removed for the Spark submit job. So, "logging enabled" in this context means, that it is enabled always for the driver and executer (hence the static "true" parameter) and that it is always disabled for the submit job (the static "false" parameter).

Should we then only run `containerdebug` in the `driver` and `executor` pods ?

Reply:
I think that it makes sense to run it also on the submit pod. You could also mount the log volume and write the log file there. But no Vector agent can be currently configured to pick this log file up, in case you planned this.